src/geometry/manifold/smooth_manifold_with_corners.lean

```diff
@@ -418,6 +438,40 @@ begin
     (of_set_mem_times_cont_diff_groupoid n I e.open_target) this
 end
 
+variables {E' : Type*} [normed_group E'] [normed_space 𝕜 E'] {H' : Type*} [topological_space H']
+
+@[simp] lemma model_with_corners_prod_coe
+  (I : model_with_corners 𝕜 E H) (I' : model_with_corners 𝕜 E' H') :
+  (I.prod I' : _ × _ → _ × _) = (prod.map I I') := rfl
```

Comment:
As above, instead of a simp lemma for the coercion, I would register a simp lemma for the local equiv product, and then other simp lemmas should fire up and do all the dirty work for you.

Reply:
I am not too sure what you mean here (I probably do not understand what's really happening). Is not lemma "model_with_corners_prod_to_local_equiv" what you are talking about?

Reply:
You're right, and it doesn't imply this one. So this one is absolutely a good idea, and the next one also. Maybe move them right after `model_with_corners_prod_to_local_equiv` because they are really of the same flavor.